### PR TITLE
Add validation of fields in log file for traffic_logstats

### PR DESCRIPTION
**Pull request by bryancall**
@dyrock I have `pqsn` in the test now.  If this commit goes in before yours it will need to be replaced by `shn`.

### CODE REVIEW

File: src/traffic_logstats/logstats.cc [hunk 1]
```diff
@@ -1275,6 +1277,19 @@ parse_log_buff(LogBufferHeader *buf_header, bool summary = false, bool aggregate
     bool agg = false;
     LogFormat::parse_symbol_string(buf_header->fmt_fieldlist(), fieldlist, &agg);
   }
+
+  // Validate the fieldlist
+  field                                                  = fieldlist->first();
+  constexpr std::array<std::string_view, 11> test_vector = {"cqtq", "ttms", "chi",  "crc", "pssc", "psql",
```
Comment (SolidWallOfCode) [inline src/traffic_logstats/logstats.cc]: Try putting more braces around the initializer. For some reason the CentOS and FreeBSD builds require this. See [here](https://github.com/SolidWallOfCode/trafficserver/blob/master/lib/ts/BufferWriterFormat.cc#L765).

Comment (bryancall) [reply src/traffic_logstats/logstats.cc]: thx - trying it out.

**Comment by SolidWallOfCode** [comment]
The compiler is unhappy about using `constexpr`. This means it can't call `traits::length` which is effectively `strlen`. You will need to reduce it to just `const`.

**Comment by bryancall** [comment]
`constexpr` works with gcc 8.1.1.

**Comment by SolidWallOfCode** [comment]
Interesting. For another PR I made my own literal string initializer for `string_view` which was `constexpr` compatible and trivial to implement. I wonder why gcc found it challenging.

**Pull request merged**